datumaro/components/hl_ops/__init__.py

```diff
@@ -0,0 +1,231 @@
+# Copyright (C) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import inspect
+import os
+import os.path as osp
+import shutil
+from typing import Dict, Optional, Type, Union
+
+from datumaro.components.dataset import Dataset, DatasetItemStorageDatasetView, IDataset
+from datumaro.components.environment import Environment
+from datumaro.components.exporter import Exporter
+from datumaro.components.filter import XPathAnnotationsFilter, XPathDatasetFilter
+from datumaro.components.launcher import Launcher, ModelTransform
+from datumaro.components.merge.exact_merge import ExactMerge
+from datumaro.components.transformer import Transform
+from datumaro.components.validator import TaskType, Validator
+from datumaro.util import parse_str_enum_value
+from datumaro.util.scope import on_error_do, scoped
+
+__all__ = ["HLOps"]
+
+
+class HLOps:
+    """High-level dataset operations for Python API."""
+
+    @staticmethod
+    def transform(
+        dataset: IDataset,
+        method: Union[str, Type[Transform]],
+        *,
+        env: Optional[Environment] = None,
+        **kwargs,
+    ) -> IDataset:
+        """
+        Applies some function to dataset items.
+
+        Results are computed lazily, if the transform supports this.
+
+        Args:
+            dataset: The dataset to be transformed
+            method: The transformation to be applied to the dataset.
+                If a string is passed, it is treated as a plugin name,
+                which is searched for in the environment
+                set by the 'env' argument
+            env: A plugin collection. If not set, the built-in plugins are used
+            **kwargs: Parameters for the transformation
+
+        Returns: a wrapper around the input dataset
+        """
+
+        if isinstance(method, str):
+            if env is None:
+                env = Environment()
+            method = env.transforms[method]
+
+        if not (inspect.isclass(method) and issubclass(method, Transform)):
+            raise TypeError("Unexpected 'method' argument type: %s" % type(method))
```

Comment:
How about to use f-string?
```suggestion
            raise TypeError(f"Unexpected 'method' argument type: {type(method)}")
```

Reply:
Replaced it with f-string.